best-practices/tech_lead.md

```diff
@@ -0,0 +1,313 @@
+
+The TL may not be able to code much
+- more devs --> more TL time needed coordinating
+- never take blocking tickets
+  - it's more important for you to be available for pairing, consulting with PO, ticket writing, board grooming, etc.
+- take tiny tickets so they won't languish half done
+- take unfun tickets so other devs can be happier
+  - e.g. review who has access to endpoints: this can be done in small pieces, and no one will be excited to do it
+
+_“What am I uniquely capable of doing?”_ - do these things; leave the rest on the board for others.  As TL, you are in a key position for information flow. Honor that via ticket writing, board grooming, pairing, etc.
+
+It helps to know who you are, what your skills and talents are, and to ask for help when you need it.
+
+## Project Management Has to Be Done
+
+... and team members are happier if it is done well.
+
+### It's About Process
+There needs to be a transparent process that:
+- documents the work cycle and sprint goals for easy lookup (e.g epics, milestones)
+- enables devs to easily choose the highest priority work for the sprint and work cycle (e.g ready column)
+- keeps all devs actively engaged in working towards sprint and work cycle goals (e.g. population of ready column, easy way for devs working on a ticket to surface problems and get resolution quickly)
+- makes it easy for the Tech Lead (TL), the Product Owner (PO) and management to see which work is in progress and how it relates to the goals (the board as a whole)
+
+If you're TL, and the work-cycle PO is mostly unavailable, and there's no appointed scrum master ... project management still needs to be done.  You don't have to do everything yourself ... but you need to ensure it all gets done.
+
+There's also engineering management opportunities when you are TL:
+- Which tech debt should we be paying?
+- “How should CDL fit into ETD work cycle?”
+- “Can the team be more-or-less single focused on ETD work for x weeks, then be single focused on Sinopia afterwards?”
+
+Ask the questions, start the discussions that will help the work-cycle and the team be as successful and happy as possible.
+
+### The Board the Board the Board
+We've been using "project boards" for transparency of process during workcycles.  More on them below.
+
+### Keep Track of Action Items Somewhere
+
+Action Items may come up in a variety of contexts:
+- during standup / discussions after standup
+- during meetings between TL and PO, or any other subset of the team
+- during full team meetings (planning, retro)
+- conversations during pairing ...
+- planned and impromptu meetings
+
+For ETD 2020Q3 work cycle, many action items ultimately became tickets, but they often started out as text in the running notes document, assigned to an individual with a comment.  Sometimes the action item was "create a ticket about" because writing the ticket needed a few more minutes than were available at the time.
```

Comment:
I would remove specific project and date mentions as it tends to age the document too quickly. In five years the specific project will be a distant memory but the advice for TL here would still be relevant.

Reply:
I'm indifferent to including project info - I don't think the aging will really be an issue here... However, I think `For example, during the 2020 ETD work cycle....` might make it read more descriptive and less prescriptive.

Reply:
added "for example" - thank you.